inode/inode.go

```diff
@@ -351,13 +352,78 @@ func (vS *volumeStruct) inodeCacheTouch(inode *inMemoryInodeStruct) {
 	vS.Unlock()
 }
 
+// TODO: What happens if the volume is removed?  What if
+// failover?
+func (vS *volumeStruct) inodeCacheDiscard() {
+	inodesToDrop := uint64(0)
+
+	vS.Lock()
+
+	if (vS.inodeCacheLRUItems * globals.inodeSize) > vS.inodeCacheLRUMaxBytes {
+		// Check, at most, 1.25 * (minimum_number_to_drop)
+		inodesToDrop = (vS.inodeCacheLRUItems * globals.inodeSize) - vS.inodeCacheLRUMaxBytes
+		inodesToDrop = inodesToDrop / globals.inodeSize
+		inodesToDrop += inodesToDrop / 4
+	}
+	for ((vS.inodeCacheLRUItems * globals.inodeSize) > vS.inodeCacheLRUMaxBytes) &&
+		(inodesToDrop > 0) {
+		inodesToDrop--
+
+		ic := vS.inodeCacheLRUHead
+
+		// Create a DLM lock object
+		id := dlm.GenerateCallerID()
+		inodeRWLock, _ := InitInodeLock(vS.volumeName, ic.InodeNumber, id)
+		err := inodeRWLock.TryWriteLock()
+
+		// Inode is locked; skip it
+		if err != nil {
+			// Move inode to tail of LRU
+			vS.inodeCacheTouchWhileLocked(ic)
+			continue
+		}
+
+		if (ic.inFlightLogSegmentMap != nil) && (ic.dirty == false) {
+			// The inode is busy - drop the DLM lock and move to tail
+			inodeRWLock.Unlock()
+			vS.inodeCacheTouchWhileLocked(ic)
+			continue
+		}
+
+		vS.Unlock()
+
+		var ok bool
+
+		ok, err = vS.inodeCacheDrop(ic)
```

Comment:
This, I think, is awkward... you could simply drop ic while still locked... that way, you'll always be dropping the head element. That said, since ...Drop() can drop out of the middle of the LRU, this is actually fine.

Reply:
Let me think this one over some more - to be honest I think it is cleaner to keep the lock and call inodeCacheDropLocked()...